### ISSUE TITLE
error: unrecognized arguments: -k WT True

### ISSUE THREAD

**Issue by catsargent** [fairliereese/LR-splitpipe#11]
I would like to use LR-splitpipe but have encountered a problem when running it as follows:

```
python /projects/b1177/software/LR-splitpipe/LR-splitpipe/demultiplex.py find_bcs \
 -f ${inDir}/m54328U_220930_232205.hifi_reads.fastq.gz \
 -o ${outDir} \
 -t 8 \
 -k WT \
 --chunksize 10**5 \
 --max_linker_dist 200 \
 --verbosity 2 \
 --delete_input True
```

Getting the following error:

```
usage: demultiplex.py [-h] {all,score_linkers,find_bcs,process_bcs} ...
demultiplex.py: error: unrecognized arguments: -k WT True

```

Can you shed any light on why I am getting this error and how to get around it?
Thanks!

**Comment by fairliereese** [comment]
Hi there! The kit identity for the barcodes used is only necessary for the barcode correction, which isn't run in the `find_bcs` module. If you run `python demultiplex.py find_bcs --help`, you'll see the options that this module takes specifically!

**Comment by catsargent** [comment]
Thank you so much for the quick response. Much appeciated!

I am now running the script as follows:

```
python /projects/b1177/software/LR-splitpipe/LR-splitpipe/demultiplex.py find_bcs \
 -f ${inDir}/m54328U_220930_232205.hifi_reads.fastq.gz \
 -o ${outDir} \
 -t 8 \
 --chunksize 100000 \
 --max_linker_dist 200 \
 --verbosity 2 \
 --delete_input
```

Note, I had to specify the chunksize as there was an error when this was not set. However, now I get the  following error:

```
Traceback (most recent call last):
  File "/projects/b1177/software/LR-splitpipe/LR-splitpipe/demultiplex.py", line 363, in <module>
    if __name__ == '__main__': main()
  File "/projects/b1177/software/LR-splitpipe/LR-splitpipe/demultiplex.py", line 325, in main
    kit = args.kit
AttributeError: 'Namespace' object has no attribute 'kit'
```
Here is the relevant code in demultiplex.py which is raising the error (from line 323):

```
	if mode == 'all' or mode == 'find_bcs' or mode == 'score_linkers':
		fastq = args.fastq
		kit = args.kit
```

So it appears that even though 'kit' is not an accepted argument when using 'find_bcs' mode, later on in the code, it is expected to have been set using args.

**Comment by catsargent** [comment]
I also got another error which was due to this:

```
		if args.max_len:
			max_len = int(args.max_len)
		else:
			max_dist = None
```

At line 333 'max_dist' should be changed to 'max_len'.

**Comment by fairliereese** [comment]
Thanks for the catch. Fixed in latest commit 376fcb7322bf52ee75005d335da02539bddc4961.
